components/UserDashboard/ProductionButton.jsx

```diff
@@ -1,17 +1,47 @@
 import { View, Text, TouchableOpacity } from "react-native";
+import ProfilePicture from "../ProfileElements/ProfilePicture";
 
 export default function ProductionButton({ navigation, production }) {
-    let participantString = `${Object.keys(production.participants).length} participant${Object.keys(production.participants).length === 1 ? "" : "s"}`;
-    
+    const participantString = `${Object.keys(production.participants).length} participant${
+        Object.keys(production.participants).length === 1 ? "" : "s"
+    }`;
     return (
-        <TouchableOpacity onPress={() => navigation.navigate("ProductionDashboard", {playCode: production.playCode})} className="w-max bg-slate-200 flex-col justify-between p-4 h-36 rounded-lg mt-3 border-2 ">
+        <TouchableOpacity
+            onPress={() =>
+                navigation.navigate("ProductionDashboard", { playCode: production.playCode })
+            }
+            className="w-max bg-slate-200 flex-col justify-between p-4 h-36 rounded-lg mt-3 border-2 "
+        >
             <Text className="text-2xl font-extrabold text-ellipsis" numberOfLines={1}>
                 {production.playName}
             </Text>
             <Text className="text-xl font-semibold">{participantString}</Text>
-            <View className="flex-row gap-1">
-                {Object.keys(production.participants).map((id) => {
-                return <View className={`w-10 h-10 bg rounded-full ${["bg-orange-400", "bg-lime-400", "bg-teal-400", "bg-fuchsia-400"][Math.floor(Math.random() * 4)]}`} key={id}></View>})}
+            <View className="w-max flex flex-row mt-2">
+                {Object.keys(production.participants)
+                    .slice(0, 5)
+                    .map((id) => {
+                        return (
+                            <>
+                                <ProfilePicture
+                                    key={id}
+                                    dimensions={10}
+                                    textSize="2xl"
+                                    userId={id}
+                                    className="mx-10"
+                                    loadingSize="small"
+                                />
+                                <View className="w-1" />
+                            </>
+                        );
+                    })}
+
+                {Object.keys(production.participants).length > 5 && (
+                    <View className="w-10 h-10 rounded-full bg-white justify-center items-center border-2 border-black">
+                        <Text className="font-semibold">
+                            {`+${Object.keys(production.participants).length - 5}`}
```

Comment:
we use the length of participants a lot so maybe define it at the top (or pass in?)